src/tools/fuzzing/fuzzing.cpp

```diff
@@ -1826,6 +1826,12 @@ void TranslateToFuzzReader::recombine(Function* func) {
 // be placed as either of the children of the i32.add. This tramples the
 // existing content there. Returns true if we found a place.
 static bool replaceChildWith(Expression* expr, Expression* with) {
+  // Casts can get broken if we replace with something not castable.
+  if (!with->type.isCastable()) {
+    if (expr->is<RefCast>() || expr->is<RefTest>() || expr->is<BrOn>()) {
+      return false;
+    }
+  }
```

Comment:
We never have an uncastable type being a subtype of a castable type, so I think this extra check shouldn't be necessary. Am I missing some edge case?

Reply:
Hmm, but we also don't have any uncastable types being supertypes of other types (except unreahcable?). I'm just worried that this is masking a more fundamental bug.